arch/risc-v/src/mpfs/Make.defs

```diff
@@ -62,14 +62,23 @@ ifeq ($(CONFIG_MPFS_DMA),y)
 CHIP_CSRCS += mpfs_dma.c
 endif
 
-ifeq ($(CONFIG_BUILD_PROTECTED),y)
-CMN_CSRCS += riscv_task_start.c
-CMN_CSRCS += riscv_pthread_start.c
-CMN_CSRCS += riscv_signal_dispatch.c
+# Flag to enable certain modules for non-flat builds (PROTECTED/KERNEL)
+__MPFS_BUILD_NONFLAT =
 
+ifeq ($(CONFIG_BUILD_PROTECTED),y)
```

Comment:
why not check ifneq ($(CONFIG_BUILD_FLAT),y) and remove __MPFS_BUILD_NONFLAT.

Reply:
My original intent was to not couple BUILD_FLAT here, but I can remote the nonflat-flag also.